test/testinput/prepbufr_adpsfc_api.py

```diff
@@ -210,5 +212,24 @@ def test_bufr_to_ioda():
    print("end")
 
 if __name__ == '__main__':
-   test_bufr_to_ioda()
+    parser = argparse.ArgumentParser()
+    description=(
+            'Reads NCEP PREPBUFR formated ADPsurface input files'
+            ' convert into IODA formatted output files. '
+    )
+
+    DATA_PATH = './testinput/gdas.t12z.adpsfc.prepbufr'
+    OUTPUT_PATH = './testrun/prepbufr_adpsfc_api.nc'
+
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument('-f', '--filename', type=str, default=DATA_PATH,
+                          dest='filename', help='adpsfc file name')
+    optional.add_argument('-o', '--output', type=str, default=OUTPUT_PATH,
+                          dest='output', help='output filename')
+    optional.add_argument('-d', '--date', type=str, default='2021 08 01 00 00',
+                          dest='date', metadatavar='YYYY mm dd HH MM', help='enable verbose debug messages')
```

Comment:
Sorry, I was editing my suggestion above while you were adding them to the branch.
```suggestion
                          dest='date', metavar='YYYY mm dd HH MM', help='analysis cycle date'
```